crates/net/network/src/state.rs

```diff
@@ -176,30 +178,37 @@ where
 
         let number = msg.block.block.header.number;
         let mut count = 0;
-        for (peer_id, peer) in self.active_peers.iter_mut() {
-            if peer.blocks.contains(&msg.hash) {
-                // skip peers which already reported the block
-                continue
-            }
 
-            // Queue a `NewBlock` message for the peer
-            if count < num_propagate {
-                self.queued_messages
-                    .push_back(StateAction::NewBlock { peer_id: *peer_id, block: msg.clone() });
+        // Shuffle peers to propagate to a random sample of peers on every block announcement
+        let mut peer_ids: Vec<PeerId> = self.active_peers.keys().cloned().collect();
+        peer_ids.shuffle(&mut rand::thread_rng());
 
-                // update peer block info
-                if self.state_fetcher.update_peer_block(peer_id, msg.hash, number) {
-                    peer.best_hash = msg.hash;
+        for peer_id in peer_ids.iter() {
+            if let Some(peer) = self.active_peers.get_mut(peer_id) {
```

Comment:
Slightly different here from the comment in the PR:

Opted for `if let` here instead of `expect` to avoid the potential panic. I don't know if the element of the hash_map could actually change while the function is running but I thought this would be slightly more robust.

Reply:
Done! Thanks for pointing this out, I learned some new Rust syntax there. :)